File: Minis/Initials.java

/*

	Program: Initials;
	Author: Badr Choubai;
	Date: 2019-08-30;
    Description: A program that takes a user's full first, middle (if they have one) and, last name. It will return the user's initials;

    Input: User's full name;
    Output: User's initials;
    
*/

import java.util.Scanner;

class Initials {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String initials = "";
        String[] fullName = { "first", "middle", "last" };

        for (String namePart : fullName) {
            System.out.printf("What is your %s name? \n", namePart);
            String answer = scanner.nextLine();
            if (answer.length() >= 1) {
                initials += answer.substring(0, 1);
            }
        }
        
        scanner.close();

        System.out.println(initials.toUpperCase());
    };
}

File: Minis/initials.py
"""

    Program: Initials 
    Description: A program that takes a user's full first, middle (if they have one) and, last name. It will return the user's initials.
    Input: User's full name 
    Output: User's initials 

"""


def get_user_input(prompts: list) -> list:
    prompt_answers: list = []

    for prompt_message in prompts:
        answer = input(f"{ prompt_message }: ")
        prompt_answers.append(answer.strip())

    return prompt_answers


def main():
    prompts: list = ["What is your first name?",
                     "What is your middle name?", "What is your last name?"]

    [first_name, middle_name, last_name] = get_user_input(prompts)

    print(f"{ last_name.title() }, { first_name.title() } { middle_name or  '' }")
    print(
        f"{ first_name[0].upper() }{ middle_name[0].upper() if middle_name else '' }{ last_name[0].upper() }")


if __name__ == "__main__":
    main()
